### PR TITLE
Add TransactionStatusEvents

### LINKED ISSUE #144: Improve tracking of Transaction status
Create a table to hold temporal data for transaction status changes. (ie: Status, ValidFromDate, ValidToDate)

Update logic or create a table trigger to maintain this temporal data. (@srkirkland What's your take on table triggers?)

### PR DESCRIPTION
**Pull request by sprucely**
Setter for `Transaction.Status` is now private to force usage of `Transaction.SetStatus()` which creates a new `TransactionStatusEvent` with details about when and where in code Status was set. (I ❤️ `[CallerMemberName]` and friends) 

closes #144

### CODE REVIEW

File: Sloth.Core/Domain/Transaction.cs [hunk 1]
```diff
@@ -25,7 +29,8 @@ public Transaction()
         [NotMapped]
         public string CreatorName => Creator?.UserName;
 
-        public string Status { get; set; }
+        // Status updates must go through SetStatus to ensure StatusEvents are properly updated
+        public string Status { get; private set; }
```
Comment (jSylvestre) [inline Sloth.Core/Domain/Transaction.cs:33]: @sprucely Dumb question... Is this going to get set when read from the db? I'm assuming yes,  but I can't remember.

Comment (sprucely) [reply Sloth.Core/Domain/Transaction.cs:33]: Yep. EF can write to private members.

File: Sloth.Sql/dbo/Tables/TransactionStatusEvents.sql [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+CREATE NONCLUSTERED INDEX [IX_TransactionStatusEvents_Status] ON [dbo].[TransactionStatusEvents] ([Status] ASC)
+
+GO
+
+CREATE NONCLUSTERED INDEX [IX_TransactionStatusEvents_EventDate] ON [dbo].[TransactionStatusEvents] ([EventDate] ASC)
+
+GO
+
+CREATE NONCLUSTERED INDEX [IX_TransactionStatusEvents_EventDetails] ON [dbo].[TransactionStatusEvents] ([EventDetails] ASC)
```
Comment (srkirkland) [inline Sloth.Sql/dbo/Tables/TransactionStatusEvents.sql:25]: Do we really need indexes on every field?  obviously the transactionID is good as a PK but I'm not sure what scenario it's useful to be able to search on the details or even date

**Review by srkirkland — APPROVED** [review]
fun implementation option and just the right amount of clever with those crazy compiler attrs

**Review by jSylvestre — APPROVED** [review]

File: Sloth.Sql/dbo/Tables/TransactionStatusEvents.sql (hunk 2 above)
Comment (sprucely) [reply Sloth.Sql/dbo/Tables/TransactionStatusEvents.sql:25]: @srkirkland the index on details made more sense before I changed what it was storing. I'll remove it. Kinda want to keep the one on EventDate though.

**Pull request merged**